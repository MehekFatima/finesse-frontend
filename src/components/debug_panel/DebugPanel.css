.radio-option input[type="radio"] {
  margin-right: 5px;
  cursor: pointer;
}
