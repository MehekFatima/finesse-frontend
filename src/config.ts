export const config = {
  backendUrl: import.meta.env.VITE_BACKEND_URL,
  debugMode: import.meta.env.VITE_DEBUG_MODE?.toLowerCase() === "true",
  githubApiUrl:
    import.meta.env.VITE_GITHUB_API_URL ||
    "https://api.github.com/repos/ai-cfia/finesse-data/contents",
  searchSource: import.meta.env.VITE_SEARCH_SOURCE || "azure",
};
