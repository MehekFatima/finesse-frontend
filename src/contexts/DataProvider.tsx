import { useState, type ReactNode } from "react";
import { config } from "../config";
import { SearchSource, type QueryResult } from "../types";
import { DataContext } from "./dataContext";

// Create a provider component
const DataProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [searchTerm, setSearchTerm] = useState<string | null>(null);
  const envSearchSource =
    config.searchSource?.toLowerCase() ?? SearchSource.azure;
  const [currentSearchSource, setCurrentSearchSource] = useState<SearchSource>(
    SearchSource[envSearchSource as keyof typeof SearchSource]
  );
  const [queryResult, setQueryResult] = useState<QueryResult[]>([]);

  return (
    <DataContext.Provider
      value={{
        searchTerm,
        setSearchTerm,
        currentSearchSource,
        setCurrentSearchSource,
        queryResult,
        setQueryResult,
      }}
    >
      {children}
    </DataContext.Provider>
  );
};

export default DataProvider;
