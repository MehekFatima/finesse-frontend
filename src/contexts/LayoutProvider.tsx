import React, { useEffect, useState, type ReactNode } from "react";
import { config } from "../config";
import { LayoutContext } from "./layoutContext";

// Create a provider component
const LayoutProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [isDebugPanelVisible, setIsDebugPanelVisible] = useState(false);
  const [isDebugButtonVisible, setIsDebugButtonVisible] = useState(false);
  const [isAlertBannerVisible, setIsAlertBannerVisible] = useState(false);

  useEffect(() => {
    setIsDebugButtonVisible(config.debugMode);
  }, []);

  const toggleDebugPanel = (): void => {
    console.log("Toggling Debug Panel visibility");
    setIsDebugPanelVisible(!isDebugPanelVisible);
  };

  return (
    <LayoutContext.Provider
      value={{
        isDebugPanelVisible,
        toggleDebugPanel,
        isDebugButtonVisible,
        isAlertBannerVisible,
        setIsAlertBannerVisible,
      }}
    >
      {children}
    </LayoutContext.Provider>
  );
};

export default LayoutProvider;
