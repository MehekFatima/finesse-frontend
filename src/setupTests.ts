// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import { config } from "./config";

export const setupTestEnvVars = (): void => {
  config.backendUrl = "backend";
  config.debugMode = true;
  config.githubApiUrl = "github";
};
