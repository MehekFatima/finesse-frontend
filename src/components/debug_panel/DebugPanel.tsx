import * as React from "react";
import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import { fetchFilenames } from "../../api/useApiUtil";
import { useData } from "../../contexts/DataContext";
import { useLayout } from "../../contexts/LayoutContext";
import { SearchSource } from "../../types";
import {
  DebugPanelContainer,
  RadioContainer,
  RadioLabel,
  RadioOption,
} from "../../styles/indexElements";
import "./DebugPanel.css";

interface DebugPanelProps {
  onSetSimulatedData?: (data: any) => void;
}

export const DebugPanel: React.FC<DebugPanelProps> = () => {
  const { isDebugPanelVisible } = useLayout();
  const { currentSearchSource, setCurrentSearchSource, setSearchTerm } =
    useData();
  const [filenames, setFilenames] = useState<string[]>([]);
  const navigate = useNavigate();

  useEffect(() => {
    if (currentSearchSource === SearchSource.static) {
      const loadFilenames = async (): Promise<void> => {
        const fetchedFilenames = await fetchFilenames();
        setFilenames(fetchedFilenames);
      };

      loadFilenames()
        .then(() => {
          console.log("Filenames loaded successfully");
        })
        .catch((error) => {
          console.error("Error loading filenames", error);
        });
    } else {
      setFilenames([]);
    }
  }, [currentSearchSource]);

  const handleFilenameClick = (
    e: React.MouseEvent<HTMLButtonElement>,
    filename: string,
  ): void => {
    e.preventDefault();
    // TODO: revisit navigate.
    navigate("/search");
    setSearchTerm(filename);
    alert("Search query set successfully!");
  };

  if (!isDebugPanelVisible) return null;
  return (
    <DebugPanelContainer data-testid="debug-panel">
      <h4 style={{ color: "black" }}>Debug Panel</h4>
      <RadioContainer>
        <RadioOption>
          <input
            data-testid="search-source-ailab"
            type="radio"
            value="ailab"
            checked={currentSearchSource === SearchSource.ailab}
            onChange={() => {
              setCurrentSearchSource(SearchSource.ailab);
            }}
          />
          <RadioLabel>Use AI Lab search</RadioLabel>
        </RadioOption>
        <RadioOption>
          <input
            data-testid="search-source-azure"
            type="radio"
            value="azure"
            checked={currentSearchSource === SearchSource.azure}
            onChange={() => {
              setCurrentSearchSource(SearchSource.azure);
            }}
          />
          <RadioLabel>Use Azure AI Search</RadioLabel>
        </RadioOption>
        <RadioOption>
          <input
            data-testid="search-source-static"
            type="radio"
            value="simulated"
            checked={currentSearchSource === SearchSource.static}
            onChange={() => {
              setCurrentSearchSource(SearchSource.static);
            }}
          />
          <span className="radio-label">Use Simulated Data</span>
        </RadioOption>
      </RadioContainer>
      {currentSearchSource === SearchSource.static && (
        <div className="input-container">
          <div>
            <h5>Filenames:</h5>
            <ul>
              {filenames.map((filename, index) => (
                <li key={index}>
                  <button
                    onClick={(e) => {
                      handleFilenameClick(e, filename);
                    }}
                  >
                    {filename}
                  </button>
                </li>
              ))}
            </ul>
          </div>
        </div>
      )}
    </DebugPanelContainer>
  );
};
