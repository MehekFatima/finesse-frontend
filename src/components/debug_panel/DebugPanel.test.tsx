import { render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { config } from "../../config";
import DataProvider from "../../contexts/DataProvider";
import LayoutProvider from "../../contexts/LayoutProvider";
import { setupTestEnvVars } from "../../setupTests";
import { DebugPanel } from "./DebugPanel";

vi.mock("react-router-dom", async () => {
  const originalModule = await vi.importActual("react-router-dom"); // Use async import if necessary
  return {
    ...originalModule,
    useNavigate: () => vi.fn(),
    // Ensure BrowserRouter is correctly exported if not automatically included
    BrowserRouter: originalModule.BrowserRouter,
  };
});

vi.mock("../../api/useApiUtil", () => ({
  fetchFilenames: async () =>
    await Promise.resolve(["file1", "file2", "file3"]),
}));

vi.mock("../../contexts/LayoutProvider", () => ({
  default: ({ children }: { children: React.ReactNode }) => (
    <div>{children}</div>
  ),
}));

vi.mock("../../contexts/layoutContext", () => ({
  useLayout: () => ({ isDebugPanelVisible: true }),
}));

describe("DebugPanel Component Tests", () => {
  const renderDebugPanel = (): HTMLElement => {
    render(
      <BrowserRouter>
        <DataProvider>
          <LayoutProvider>
            <DebugPanel />
          </LayoutProvider>
        </DataProvider>
      </BrowserRouter>
    );
    return screen.getByTestId("debug-panel");
  };
  const originalEnv = process.env;

  beforeAll(() => {
    setupTestEnvVars();
    window.alert = vi.fn();
  });

  afterEach(() => {
    vi.resetAllMocks();
  });

  afterAll(() => {
    process.env = originalEnv;
  });

  it("renders radio buttons for search sources", async () => {
    const view = renderDebugPanel();
    await waitFor(() => {
      const radioButtons = screen.queryAllByRole("radio");
      expect(radioButtons.length).toBe(2);
    });
    expect(view).toMatchSnapshot();
  });


  it('selects the "Azure AI" radio button when VITE_SEARCH_SOURCE is azure', async () => {
    config.searchSource = "azure";
    const view = renderDebugPanel();
    const azureRadioButton = screen.getByTestId("search-source-azure");
    const llamaindexRadioButton = screen.getByTestId(
      "search-source-llamaindex"
    );
    expect(azureRadioButton).toBeInTheDocument();
    expect(llamaindexRadioButton).toBeInTheDocument();
    expect(azureRadioButton).toBeChecked();
    expect(llamaindexRadioButton).not.toBeChecked();
    expect(view).toMatchSnapshot();
  });

  it('selects the "AI Lab LlamaIndex" radio button when VITE_SEARCH_SOURCE is llamaindex', async () => {
    config.searchSource = "llamaindex";
    const view = renderDebugPanel();
    const azureRadioButton = screen.getByTestId("search-source-azure");
    const llamaindexRadioButton = screen.getByTestId(
      "search-source-llamaindex"
    );
    expect(azureRadioButton).toBeInTheDocument();
    expect(llamaindexRadioButton).toBeInTheDocument();
    expect(azureRadioButton).not.toBeChecked();
    expect(llamaindexRadioButton).toBeChecked();
    expect(view).toMatchSnapshot();
  });

  it("selects no radio button when VITE_SEARCH_SOURCE has a bad value", async () => {
    config.searchSource = "bad";
    const view = renderDebugPanel();
    const azureRadioButton = screen.getByTestId("search-source-azure");
    expect(azureRadioButton).not.toBeChecked();
    expect(view).toMatchSnapshot();
  });
});
