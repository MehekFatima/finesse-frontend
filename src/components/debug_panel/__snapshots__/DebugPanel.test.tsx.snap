// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DebugPanel Component Tests > can switch to simulated data and display filenames 1`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
  <div
    class="input-container"
  >
    <div>
      <h5
        style="color: black;"
      >
        Filenames:
      </h5>
      <ul>
        <li>
          <button>
            file1
          </button>
        </li>
        <li>
          <button>
            file2
          </button>
        </li>
        <li>
          <button>
            file3
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`DebugPanel Component Tests > clicking on a filename button updates the search term 1`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
  <div
    class="input-container"
  >
    <div>
      <h5
        style="color: black;"
      >
        Filenames:
      </h5>
      <ul>
        <li>
          <button>
            file1
          </button>
        </li>
        <li>
          <button>
            file2
          </button>
        </li>
        <li>
          <button>
            file3
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`DebugPanel Component Tests > clicking on a filename button updates the search term 2`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
  <div
    class="input-container"
  >
    <div>
      <h5
        style="color: black;"
      >
        Filenames:
      </h5>
      <ul>
        <li>
          <button>
            file1
          </button>
        </li>
        <li>
          <button>
            file2
          </button>
        </li>
        <li>
          <button>
            file3
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`DebugPanel Component Tests > clicking on a filename button updates the search term 3`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
  <div
    class="input-container"
  >
    <div>
      <h5
        style="color: black;"
      >
        Filenames:
      </h5>
      <ul>
        <li>
          <button>
            file1
          </button>
        </li>
        <li>
          <button>
            file2
          </button>
        </li>
        <li>
          <button>
            file3
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`DebugPanel Component Tests > no radio button is selected when REACT_APP_SEARCH_SOURCE has a bad value 1`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
</div>
`;

exports[`DebugPanel Component Tests > renders radio buttons for search sources 1`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
</div>
`;

exports[`DebugPanel Component Tests > selects the "AI Lab" radio button when REACT_APP_SEARCH_SOURCE is ailab 1`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        checked=""
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
</div>
`;

exports[`DebugPanel Component Tests > selects the "Azure AI" radio button when REACT_APP_SEARCH_SOURCE is azure 1`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
</div>
`;

exports[`DebugPanel Component Tests > selects the "Simulated Data" radio button when REACT_APP_SEARCH_SOURCE is static 1`] = `
<div
  class="sc-irLvIq hkeVyu"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-csKJxZ ilyJmt"
  >
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-eTNRI cNqKFl"
    >
      <input
        checked=""
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-brSamD fpMuCT"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
  <div
    class="input-container"
  >
    <div>
      <h5
        style="color: black;"
      >
        Filenames:
      </h5>
      <ul />
    </div>
  </div>
</div>
`;

exports[`DebugPanel Component Tests can switch to simulated data and display filenames 1`] = `
<div
  class="sc-iqHXzD heCJM"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-crrtmM KqfJY"
  >
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
  <div
    class="input-container"
  >
    <div>
      <h5
        style="color: black;"
      >
        Filenames:
      </h5>
      <ul>
        <li>
          <button>
            file1
          </button>
        </li>
        <li>
          <button>
            file2
          </button>
        </li>
        <li>
          <button>
            file3
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`DebugPanel Component Tests clicking on a filename button updates the search term 1`] = `
<div
  class="sc-iqHXzD heCJM"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-crrtmM KqfJY"
  >
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
  <div
    class="input-container"
  >
    <div>
      <h5
        style="color: black;"
      >
        Filenames:
      </h5>
      <ul>
        <li>
          <button>
            file1
          </button>
        </li>
        <li>
          <button>
            file2
          </button>
        </li>
        <li>
          <button>
            file3
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`DebugPanel Component Tests no radio button is selected when REACT_APP_SEARCH_SOURCE has a bad value 1`] = `
<div
  class="sc-iqHXzD heCJM"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-crrtmM KqfJY"
  >
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
</div>
`;

exports[`DebugPanel Component Tests renders radio buttons for search sources 1`] = `
<div
  class="sc-iqHXzD heCJM"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-crrtmM KqfJY"
  >
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
</div>
`;

exports[`DebugPanel Component Tests selects the "AI Lab" radio button when REACT_APP_SEARCH_SOURCE is ailab 1`] = `
<div
  class="sc-iqHXzD heCJM"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-crrtmM KqfJY"
  >
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        checked=""
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
</div>
`;

exports[`DebugPanel Component Tests selects the "Azure AI" radio button when REACT_APP_SEARCH_SOURCE is azure 1`] = `
<div
  class="sc-iqHXzD heCJM"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-crrtmM KqfJY"
  >
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        checked=""
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
</div>
`;

exports[`DebugPanel Component Tests selects the "Simulated Data" radio button when REACT_APP_SEARCH_SOURCE is static 1`] = `
<div
  class="sc-iqHXzD heCJM"
  data-testid="debug-panel"
>
  <h4
    style="color: black;"
  >
    Debug Panel
  </h4>
  <div
    class="sc-crrtmM KqfJY"
  >
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-ailab"
        type="radio"
        value="ailab"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use AI Lab search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        data-testid="search-source-azure"
        type="radio"
        value="azure"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Azure AI Search
      </span>
    </label>
    <label
      class="sc-dQpIV knqZxe"
    >
      <input
        checked=""
        data-testid="search-source-static"
        type="radio"
        value="simulated"
      />
      <span
        class="sc-bqyKbq jYLzpr"
      >
        Use Simulated Data
      </span>
    </label>
  </div>
  <div
    class="input-container"
  >
    <div>
      <h5
        style="color: black;"
      >
        Filenames:
      </h5>
      <ul />
    </div>
  </div>
</div>
`;
